File: ICPC Brazilian Subregional/2019/M.py
def check(seg, pipocas):
    # seg = segundos para a equipe comer todas as pipocas

    por_pessoa = seg * t
    temp = 0
    i = 0
    
    for x in range(c):
        temp = 0
        while i < len(pipocas):
            if temp + pipocas[i] > por_pessoa:
                break
            temp += pipocas[i]
            i += 1
        
        
    return i == len(pipocas)

n,c,t = [int(i) for i in input().split()]

pipocas = [int(i) for i in input().split()]


l = 1
r = 10**9+1 # ver qnt pode ser o max de segundos possivel
while l < r:
    mid = (l+r)//2

    if check(mid, pipocas):
        r = mid
    else:
        l = mid + 1
        
print(l)

File: ICPC/ICPC Brazilian Subregional 2019/2019/M.cpp
#include <bits/stdc++.h>
#define MAX 100001

using namespace std;

int n,c,t;
int pipocas[MAX];

bool check(long long seg){

    long long por_pessoa = seg * t;
    long long temp = 0;
    int i = 0;
    
    for(int x=0; x <c; x++){
    
        temp = 0;
        
        while(i < n){
            if(temp + pipocas[i] > por_pessoa) break;
            
            temp += pipocas[i];
            i++;
        }
    }
    
    return i == n;
}


int main(){


    cin >> n >> c >> t;
    for(int i=0; i<n; i++)
        cin >> pipocas[i];

    long long l=1LL;
    long long r = 10000000001;
    long long mid;
    
    while(l<r){
    
        mid = (l+r) / 2;
        
        if(check(mid))
            r = mid;
        else
            l = mid + 1;
    
    }


    cout << l << endl;

}
